From 0b4c628f53ef7ede7b78980c4080c537f5b41091 Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Thu, 25 Aug 2016 14:49:24 +0300
Subject: [PATCH 0439/1345] dts: a3700: add support for the Armada 3720
 community board

commit  d0335fdae34603e5a345529414ac85c658d27ab4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch is to add initial support for Armada 3720
community board.
- add new dts file for the Armada 3720 community board
- add Armada 3720 community board device tree file to Makefile

Change-Id: Ie4e21bde8c3574c3b95f6c1aeff6ed7ff50c855a
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32224
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 .../boot/dts/marvell/armada-3720-community.dts     |  108 ++++++++++++++++++++
 2 files changed, 109 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-3720-community.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 7adaba9..e06cf37 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -9,6 +9,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-rz-db-sfp.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sgmii1.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sfp.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-router.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
new file mode 100644
index 0000000..31838b6
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -0,0 +1,108 @@
+/*
+ * Device Tree file for Marvell Armada 3720 community board
+ * Copyright (C) 2016 Marvell
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+
+#include "armada-372x-rz.dtsi"
+
+/ {
+	model = "Marvell Armada 3720 Community Board";
+	compatible = "marvell,armada-3720-community", "marvell,armada3720", "marvell,armada3710";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
+	};
+
+	soc {
+		internal-regs {
+			uart0: serial@12000 {
+				status = "okay";
+			};
+
+			i2c0: i2c@11000 {
+				status = "okay";
+			};
+
+			i2c1: i2c@11080 {
+				status = "okay";
+			};
+
+			eth0: ethernet@30000 {
+				phy-mode = "rgmii-id";
+				status = "okay";
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+
+			usb3@58000 {
+				status = "okay";
+			};
+
+			usb@5e000 {
+				status = "okay";
+			};
+
+			sdhci1: sdhci@d0000 {
+				wp-inverted;
+				bus-width = <4>;
+				cd-gpios = <&gpio_nb 3 GPIO_ACTIVE_LOW>;
+				no-1-8-v;
+				status = "okay";
+
+				pinctrl-names = "default";
+				pinctrl-0 = <&sdio_pins &cd_pins>;
+			};
+
+			sata@e0000 {
+				status = "okay";
+			};
+		};
+	};
+};
-- 
1.7.9.5

